src/rulesets/Base/fastmath_able.jl

```diff
@@ -164,14 +164,16 @@ let
         @scalar_rule x - y (true, -1)
         @scalar_rule x / y (one(x) / y, -(Ω / y))
         #log(complex(x)) is required so it gives correct complex answer for x<0
-        @scalar_rule(x ^ y,
-            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(complex(x))),
-        )
+        @scalar_rule(x ^ y, (
```

Comment:
This  pre-existing rule is unsafe for non-real and complex numbers. e.g. this will fail for quaternions because you can't call `complex` on them and because it assumes commutativity of multiplication. I don't know if part of this PR, but the types should probably be restricted.